modules/dcache-vehicles/src/test/java/diskCacheV111/vehicles/JobInfoTests.java

```diff
@@ -0,0 +1,74 @@
+package diskCacheV111.vehicles;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.*;
+
+/**
+ * Created by alenaschemmert on 03.08.17.
+ */
+
+public class JobInfoTests {
+
+    private JobInfo jobInfo;
+    private OldJobInfo oldJobInfo;
+    private String ref;
+
+    @Before
+    public void setUp() {
+        jobInfo = new JobInfo(System.currentTimeMillis(), System.currentTimeMillis() + 1000, "working", 1, "waldo", 2407);
+        oldJobInfo = new OldJobInfo(System.currentTimeMillis(), System.currentTimeMillis() + 1000, "working", 1, "waldo", 2407);
+        ref = jobInfo.toString();
+    }
+
+    @Test
+    public void toStringShouldBeThreadSafe() {
+        Callable<String> task = jobInfo::toString;
+        ExecutorService exec = Executors.newFixedThreadPool(8);
+        List<Future<String>> results = new ArrayList<>();
+
+        for (int i = 0; i < 8; i++) {
+            results.add(exec.submit(task));
+        }
+        exec.shutdown();
+
+        for (Future<String> result : results) {
+            try {
+                Assert.assertEquals(ref, result.get());
+            } catch (InterruptedException | ExecutionException e) {
+                Assert.assertFalse(true);
+            }
+        }
+    }
+
+    @Test
+    public void OldAndNewShouldBeEqual() {
+        Assert.assertEquals(jobInfo.toString(), oldJobInfo.toString());
+    }
+
+    private static class OldJobInfo extends JobInfo {
```

Comment:
`toString()` is often designed for human consumption (e.g., interactive admin commands, log files, ...).  Writing unit-tests that are specific enough to catch problems, but general enough to allow "reasonable" changes is difficult.  There's also the very real risk that someone has written some script that parses the output.

Ultimately, we don't know what expectations our users have here, so we have to "use our best judgement".

However, to give some concrete advice here, your patch is about the output of the formatted string, so it's reasonable that your unit-test focuses on that part and ignores everything else.

Lets say you update your `@Before` method to use a fixed time.  Any value is fine, for example suppose the submit time is 1502369009000L (which is about now).  We expect that the output from `toString();` will contain this time in the correct format; i.e.,  "08/10-14:43:29".

So, the simplest test would be (using the newer `assertThat` syntax) to check whether that string appears in the output:

    assertThat(toString, containsString("08/10-14:43:29"));

If the startTime is sufficiently different (currently it is one second after submitTime), then an additional assertion can be added, looking for that string:

    assertThat(toString, containsString("08/10-14:43:30"));

Certainly, this is a very limited test, but sufficient to check that the code works as anticipated.

The assertion could be made stronger, by insisting that the output is a `;`-separated list of items, two of which must match each of the dates exactly, something like:

    List<String> toStringItems = Splitter.on(';').splitToList(toString)
    assertThat(toStringItems, contains("08/10-14:43:29"));
    assertThat(toStringItems, contains("08/10-14:43:30"));

The tests could be made stronger still by insisting in the order of these items, but I think this is roughly the right balance between being too restrictive in the unit-test and catching potential problems.